tutorial.md

```diff
@@ -0,0 +1,1140 @@
+    aswell this is no problem however; please report any
+    incompatibilities you noticed and state the version of xmonad and
+    xmonad-contrib that you tried to use.  If you use the git versions
+    of xmonad and xmonad-contrib, you should be able to follow
+    everything just fine.
+
+TODO:
+
+  * The closing thoughts could probably be a bit more extensive.
+
+  * People may be inclined to use a display manager, especially when
+    coming from proper DEs.  I have no experience with those, so help
+    would be appreciated as to what a section or paragraph mentioning
+    this should actually contain.
+
+  * Suggest alternative programs?  I feel like this could explode very
+    quickly.
+
+QUESTIONS:
+
+  * Is anything missing that beginners really ought to know about?
+
+  * Is this comprehensible for people who don't know Haskell?
+
+# XMonad Configuration Tutorial
+
+We're going to take you, step-by-step, through the process of
+configuring xmonad, setting up a status bar with [xmobar], setting up a
+tray with [trayer-srg], and making it all play nicely together.
+
+It is assumed that you have read the [xmonad guided tour] already.  It
+is a bit dated at this point but, because xmonad is stable, the guide
+should still give you a good overview of the most basic functionality.
+
+Before we begin, here are two screenshots of the kind of desktop we will
+be creating together.  In particular, a useful layout we'll conjure into
+being—the three columns layout from [XMonad.Layout.ThreeColumns] with
+the ability to magnify stack windows with [XMonad.Layout.Magnifier]:
+
+<p>
+<img alt="blank desktop" src="https://user-images.githubusercontent.com/50166980/111586872-c0eb0e00-87c1-11eb-9129-c2781bbbf227.png" width="550">
+<img alt="blank desktop" src="https://user-images.githubusercontent.com/50166980/111586885-c47e9500-87c1-11eb-8e8e-83d4d33c4c8d.png" width="550">
+</p>
+
+So let's get started!
+
+## Preliminaries
+
+First you'll want to install xmonad.  You can either do this with your
+system's package manager, or via `stack`/`cabal`.  The latter may be
+necessary if your package manager does not package `xmonad` and
+`xmonad-contrib` (or packages a version that's very old) or you want to
+use the git versions instead of a tagged release.  You can find
+instructions for how to do this in [INSTALL.md].
+
+One more word of warning: if you are on a distribution that installs
+every Haskell package dynamically linked—thus essentially breaking
+Haskell—(Arch Linux being a prominent example) then we would highly
+recommend installing via `stack` or `cabal` as instructed in
+[INSTALL.md].  If you still want to install xmonad via your system's
+package manager, you will need to `xmonad --recompile` _every time_ a
+Haskell dependency is updated.
+
+We're going to assume xmonad version `0.16` and xmonad-contrib version
+`0.17` here, though most of these steps should work with older versions
```

Comment:
Maybe also mention the git version here. I think xmonad-contrib 0.17 is not finalized. Also do we need to differentiate between 0.17 and git?

Reply:
The tutorial will be "released" in tandem with 0.17, so I think this still makes sense